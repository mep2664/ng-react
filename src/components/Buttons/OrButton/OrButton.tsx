import * as React from "react";
import styled from "styled-components"
import { ButtonType } from "../";
import { bgColor, fontColor } from "../../../theme";

const ButtonWrapper = styled.div`
    position: relative;
    width: 100%;
    display: flex;
    justify-content: center;
`;

const OrSpan = styled.span`
    position: absolute;
    left: 50%;
    margin-top: 7.5px;
    margin-left: -12.5px;
    line-height: 25px;
    height: 25px;
    width: 25px;
    text-align: center;
    color: #1f1f1f;
    background: #f0f0f0;
    border-radius: 20px;
`;

const SharedButtonStyle = `
    height: 40px;
    width: 150px;
    padding: 0 20px;
    color: #1f1f1f;
    border: none;
    font-family: Arial, "Arial Header", sans-serif;
    font-size: 16px;
    font-weight: 600;
    cursor: pointer;
`

const LeftButton = styled.button`
    ${SharedButtonStyle}
    margin-right: 1px;
    border-radius: 20px 0 0 20px;
    background-color: ${bgColor["Positive"]};
    color: ${fontColor["Positive"]};
`;

const RightButton = styled.button`
    ${SharedButtonStyle}
    margin-left: 1px;
    border-radius: 0 20px 20px 0;
    background-color: ${bgColor["Positive"]};
    color: ${fontColor["Positive"]};
`;

export interface IOrButtonProps {
    leftCaption: string;
    rightCaption: string;
    leftType?: ButtonType;
    rightType?: ButtonType;
    leftForm?: string;
    rightForm?: string;

    leftOnClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
    rightOnClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
};

export const OrButton: React.FC<IOrButtonProps> = ({ leftCaption, rightCaption, leftType, rightType, leftForm, rightForm, leftOnClick, rightOnClick }) => {
    return (
        <ButtonWrapper>
            <LeftButton
                type={leftType || "button"}
                form={leftForm}
                onClick={leftOnClick}
            >
                {leftCaption}
            </LeftButton>
            <OrSpan>or</OrSpan>
            <RightButton
                type={rightType || "button"}
                form={rightForm}
                onClick={rightOnClick}
            >
                {rightCaption}
            </RightButton>
        </ButtonWrapper>
    );
}
