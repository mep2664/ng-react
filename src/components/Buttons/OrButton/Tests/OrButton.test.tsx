import React from "react"
import { render, fireEvent } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect";
import { OrButton } from "../OrButton"
import { ButtonType } from "../../";

describe("Tests for the Button component", () => {
    const leftCaption = "LEFT_CAPTION";
    const rightCaption = "RIGHT_CAPTION";
    const leftOnClick = jest.fn();
    const rightOnClick = jest.fn();
    const leftForm = "LEFT_FORM";
    const rightForm = "RIGHT_FORM";
    const leftType: ButtonType = "submit";
    const rightType: ButtonType = "reset";

    const props = {
        leftCaption, rightCaption, leftOnClick, rightOnClick, leftForm, rightForm, leftType, rightType,
    };

    test("Component matches snapshot", () => {
        const { container } = render(
            <OrButton
                {...props}
            />,
        );
        expect(container).toMatchSnapshot();
    });

    test("Left button exists", () => {
        const { getByText } = render(
            <OrButton
                {...props}
            />,
        )
        expect(getByText(leftCaption)).toBeDefined();
    });

    test("Right button exists", () => {
        const { getByText } = render(
            <OrButton
                {...props}
            />,
        )
        expect(getByText(rightCaption)).toBeDefined();
    });

    test("Empty types attribute is set correctly", () => {
        const { getByText } = render(
            <OrButton
                {...props}
                leftType={undefined}
                rightType={undefined}
            />,
        )
        const leftButton = getByText(leftCaption);
        expect(leftButton).toHaveAttribute("type", "button");
        const rightButton = getByText(rightCaption);
        expect(rightButton).toHaveAttribute("type", "button");
    });

    test("leftType attribute is set correctly", () => {
        const { getByText } = render(
            <OrButton
                {...props}
            />,
        )
        const button = getByText(leftCaption);
        expect(button).toHaveAttribute("type", leftType);
    });

    test("rightType attribute is set correctly", () => {
        const { getByText } = render(
            <OrButton
                {...props}
            />,
        )
        const button = getByText(rightCaption);
        expect(button).toHaveAttribute("type", rightType);
    });

    test("leftForm attribute is set correctly", () => {
        const { getByText } = render(
            <OrButton
                {...props}
            />,
        )
        const button = getByText(leftCaption);
        expect(button).toHaveAttribute("form", leftForm);
    });

    test("rightForm attribute is set correctly", () => {
        const { getByText } = render(
            <OrButton
                {...props}
            />,
        )
        const button = getByText(rightCaption);
        expect(button).toHaveAttribute("form", rightForm);
    });

    test("leftOnClick event fires correctly", () => {
        const { getByText } = render(
            <OrButton
                {...props}
            />,
        )
        const button = getByText(leftCaption);
        fireEvent.click(button);
        expect(leftOnClick).toHaveBeenCalledTimes(1);
    });

    test("rightOnClick event fires correctly", () => {
        const { getByText } = render(
            <OrButton
                {...props}
            />,
        )
        const button = getByText(rightCaption);
        fireEvent.click(button);
        expect(rightOnClick).toHaveBeenCalledTimes(1);
    });
});
