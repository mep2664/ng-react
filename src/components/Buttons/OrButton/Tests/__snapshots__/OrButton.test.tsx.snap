// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the Button component Component matches snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa kOWXyZ"
  >
    <button
      class="sc-htpNat ewFkYC"
      form="LEFT_FORM"
      type="submit"
    >
      LEFT_CAPTION
    </button>
    <span
      class="sc-bwzfXH gMKPiz"
    >
      or
    </span>
    <button
      class="sc-bxivhb cPfNQV"
      form="RIGHT_FORM"
      type="reset"
    >
      RIGHT_CAPTION
    </button>
  </div>
</div>
`;
